pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -482,6 +488,82 @@ func buildFileAccessLogHelper(path string, mesh *meshconfig.MeshConfig) *accessl
 	return al
 }
 
+func buildOpenTelemetryLogHelper(pushCtx *model.PushContext,
+	provider *meshconfig.MeshConfig_ExtensionProvider_EnvoyOpenTelemetryLogProvider) *accesslog.AccessLog {
+	_, cluster, err := clusterLookupFn(pushCtx, provider.Service, int(provider.Port))
+	if err != nil {
+		log.Errorf("could not find cluster for open telemetry provider %q: %v", provider, err)
+		return nil
+	}
+
+	logName := provider.LogName
+	if logName == "" {
+		logName = otelEnvoyAccessLogFriendlyName
+	}
+
+	f := provider.Format
+	if f == "" {
+		f = EnvoyTextLogFormat
+	}
+
+	cfg := buildOpenTelemetryAccessLogConfig(logName, cluster, f, provider.Attributes)
+
+	return &accesslog.AccessLog{
+		Name:       otelEnvoyALSName,
+		ConfigType: &accesslog.AccessLog_TypedConfig{TypedConfig: util.MessageToAny(cfg)},
+	}
+}
+
+func buildOpenTelemetryAccessLogConfig(logName, clusterName, format string, attrs map[string]string) *otelaccesslog.OpenTelemetryAccessLogConfig {
+	cfg := &otelaccesslog.OpenTelemetryAccessLogConfig{
+		CommonConfig: &grpcaccesslog.CommonGrpcAccessLogConfig{
+			LogName: logName,
+			GrpcService: &core.GrpcService{
+				TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
+					EnvoyGrpc: &core.GrpcService_EnvoyGrpc{
+						ClusterName: clusterName,
+					},
+				},
+			},
+			TransportApiVersion:     core.ApiVersion_V3,
+			FilterStateObjectsToLog: envoyWasmStateToLog,
+		},
+	}
+
+	if format != "" {
+		cfg.Body = &otlpcommon.AnyValue{
+			Value: &otlpcommon.AnyValue_StringValue{
+				StringValue: format,
+			},
+		}
+	}
+
+	if len(attrs) != 0 {
+		cfg.Attributes = &otlpcommon.KeyValueList{
+			Values: convertMapToAttributeKeyValues(attrs),
+		}
+	}
+
+	return cfg
+}
+
+func convertMapToAttributeKeyValues(attrsMap map[string]string) []*otlpcommon.KeyValue {
+	if attrsMap == nil {
+		return nil
+	}
+	attrList := make([]*otlpcommon.KeyValue, len(attrsMap))
+	index := 0
+	for key, value := range attrsMap {
+		kv := otlpcommon.KeyValue{
+			Key:   key,
+			Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: value}},
+		}
+		attrList[index] = &kv
```

Comment:
nit: `append` is a bit less error prone. Just make sure you update L554 to `make([]*otlpcommon.KeyValue, 0, len(attrsMap))` if you do

Reply:
thanks, there's a mistake, will update all this after API is merged